lib/rubygems/name_tuple.rb

```diff
@@ -49,7 +47,7 @@ def self.null
 
   def full_name
     case @platform
-    when nil, "ruby", ""
+    when nil, "", Gem::Platform::RUBY
```

Comment:
this is mostly impossible given `initialize`. This should just be...
```suggestion
    if @platform == Gem::Platform::RUBY
```

Reply:
Seems it's possible that Marshal could load a NameTuple with a nil/empty platform, so I'll leave this as is.